src/main/java/core/basesyntax/classes/functional/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.classes.functional;
+
+import core.basesyntax.classes.figures.Circle;
+import core.basesyntax.classes.figures.Figure;
+import core.basesyntax.classes.figures.IsoscelesTrapezoid;
+import core.basesyntax.classes.figures.Rectangle;
+import core.basesyntax.classes.figures.RightTriangle;
+import core.basesyntax.classes.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        final int numberOfFigures = 5;
+        final int maxSideSize = 12;

```

Comment:
Consider defining a constant for the maximum side size `12` to avoid magic numbers. This will improve code readability and maintainability.